src/parser.cpp

```diff
@@ -0,0 +1,637 @@
+                             */
+                            variable2 = std::make_shared<Variable>();
+                            variable2->setName(variableName);
+                            component2->addVariable(variable2);
+                        } else {
+                            VariableErrorPtr err = std::make_shared<VariableError>();
+                            err->setDescription("Connection variable_2 '" + variableName +
+                                                "' not found in component_2 '" + component2->getName() + "'.");
+                            err->setComponent(component2);
+                            addError(err);
+                            errorOccurred = true;
+                        }
+                    } else {
+                        ModelErrorPtr err = std::make_shared<ModelError>();
+                        err->setDescription("Connection does not name a valid component_2 for connection "
+                                            "between variables variable_1='" + variable1->getName() + "' "
+                                            "and variable_2='" + variableName + "'.");
+                        err->setModel(model);
+                        addError(err);
+                        errorOccurred = true;
+                    }
+                } else {
+                    ModelErrorPtr err = std::make_shared<ModelError>();
+                    err->setDescription("Connection map_variables does not contain a variable_2.");
+                    err->setModel(model);
+                    addError(err);
+                    errorOccurred = true;
+                }
+            } else {
+                ModelErrorPtr err = std::make_shared<ModelError>();
+                err->setDescription("Connection map_variables does not contain a variable_1.");
+                err->setModel(model);
+                addError(err);
+                errorOccurred = true;
+            }
+        } else {
+            ModelErrorPtr err = std::make_shared<ModelError>();
+            err->setDescription("Invalid element type '" + mapVariablesNode->getType() +
+                                "' found in model connection. Expected a map_variables element.");
+            err->setModel(model);
+            addError(err);
+            errorOccurred = true;
+        }
+
+        // Set variable equivalence relationship.
+        if (!errorOccurred) {
+            Variable::addEquivalence(variable1, variable2);
+        }
+        // Continue on to next map_variables if one exists
+        mapVariablesNode = mapVariablesNode->getNext();
+    }
+}
+
+void Parser::loadEncapsulation(const ModelPtr &model, XmlNodePtr &parentComponentNode)
+{
+    while (parentComponentNode) {
+        bool errorOccurred = false;
+        ComponentPtr parentComponent = nullptr;
+        if (parentComponentNode->isType("component_ref")) {
+            if (parentComponentNode->hasAttribute("component")) {
```

Comment:
If so, then the `loadImport()` method (for example) shouldn't generate an error message if an attribute is neither `href` nor `xlink`. I guess it all comes down to (once again) trying to be consistent throughout the code.